### DISCUSSION TITLE
How 'reliable' is the ordering of statements in queries; deleteMany, create, set, etc.

### DISCUSSION THREAD

**Discussion by richardwardza** [prisma/prisma#6263] [Q&A] [answered]
I'd like to know if the order that you write the statements in is maintained when the statement is run, easiest shown with an example:

A User has many Attributes - When updating the User I'd like to clear any existing attributes before creating new ones

```ts
prisma.user.update({
    where: {id: 1},
    data: {
        age: 33
        attributes: {
            deleteMany: {},
            create: [
                { name: "arm", value: 2 },
                { name: "shoulders, value: 2 }
            ]
        }
    }
});
```
In the `attributes` section will the `deleteMany` be guaranteed to run before the `create`?

Obviously I don't want to create the new records and then have everything deleted.. :)

**Comment by ryands17** [comment, marked as answer]
@richardwardza 👋 

Wouldn't `set` work for you in this case?

**Reply by richardwardza** [reply]
✌️  @ryands17

`set` would work if the new new 'attributes' already exist, but in my instance they do not. 
I'd like to be able to clear any existing relations and then `create` new ones.

I could do 2 queries, the first using `set` to remove any existing relationships and then the second to create and associate the new items but I'm hoping to get it done in 1, hence the question. if the `deleteMany` will always run before the `create`.

**Reply by ryands17** [reply]
@richardwardza 👋 

In the latest version, I have found that the order matters so in your case, it would be `deleteMany` before `create` :)

It's most probably using `Object.keys` which is deterministic in the ES* versions.

**Reply by richardwardza** [reply]
Thanks @ryands17 - I found the same with my "testing" but just wanted to make sure :)

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
